app/src/main/res/layout/activity_setup.xml

```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```

Comment:
Yeah, we can replace with `FrameLayout`. Making the activity it's self a root `Navhost` mean, we will be implementing the same logic for `TheActivity` as well right? and also this would allow us to use this host view irrespective of the activity lifecycle. We can use this in `Fragment`s or `View`s it self.

Reply:
Could you please explain this part further?
> and also this would allow us to use this host view irrespective of the activity lifecycle. We can use this in Fragments or Views it self.

Reply:
Sorry, I completely misunderstood the comment. Yeah, we can use the `ScreenHostView` as root layout for the activity.